services/frontend-service/src/ui/components/ProductVersion/ReleaseTrainDialog.test.tsx

```diff
@@ -0,0 +1,221 @@
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'dev' } },
+                },
+                {
+                    name: 'test3',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'dev' } },
+                },
+            ],
+            expectedEnvs: ['test', 'test2', 'test3'],
+            filteredEnvs: ['test1'],
+        },
+        {
+            name: 'filter out all environments',
+            input: {
+                environment: 'dev',
+                open: true,
+                onCancel: myCancelSpy,
+            },
+            envsList: [
+                {
+                    name: 'test',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'staging' } },
+                },
+                {
+                    name: 'test1',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'staging' } },
+                },
+                {
+                    name: 'test2',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'staging' } },
+                },
+                {
+                    name: 'test3',
+                    locks: {},
+                    distanceToUpstream: 0,
+                    priority: 0,
+                    applications: {},
+                    config: { upstream: { environment: 'staging' } },
+                },
+            ],
+            filteredEnvs: ['test', 'test1', 'test2', 'test3'],
+            expectedEnvs: [],
+        },
+    ];
+    describe.each(data)(`Displays Product Version Table`, (testCase) => {
+        const getNode = (overrides: ReleaseTrainDialogProps) => <ReleaseTrainDialog {...overrides} />;
+        const getWrapper = (overrides: ReleaseTrainDialogProps) => render(getNode(overrides));
+        it(testCase.name, () => {
+            mock_UseEnvs.returns(testCase.envsList);
+            myCancelSpy.mockReset();
+            expect(myCancelSpy).toHaveBeenCalledTimes(0);
+            getWrapper(testCase.input);
+            testCase.expectedEnvs.forEach((value, index) => {
+                expect(documentQuerySelectorSafe('.id-' + value)).toBeDefined();
+            });
+            testCase.filteredEnvs.forEach((value, index) => {
+                expect(screen.queryByText(value)).not.toBeInTheDocument();
+            });
+        });
```

Comment:
feels like theres a line missing like
```suggestion
        expect(myCancelSpy).toHaveBeenCalledTimes(someNumberGoesHere);
        });
```

Reply:
File removed, due to redundancy :)